QEfficient/transformers/models/mixtral_moe/modeling_mixtral.py

```diff
@@ -0,0 +1,829 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c)  2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+"""PyTorch Mixtral model."""
+
+import inspect
+import math
+import warnings
+from typing import List, Optional, Tuple, Union
+import onnxscript
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+from torch import nn
+
+from transformers.activations import ACT2FN
+
+from transformers.models.mixtral.modeling_mixtral import (
+    logger,
+    MixtralAttention,
+    MixtralForCausalLM,
+    MixtralModel,
+    MixtralConfig,
+    MixtralDecoderLayer,
+    MixtralSparseMoeBlock,
+    MixtralBLockSparseTop2MLP,
+    MixtralRotaryEmbedding,
+    MixtralRMSNorm,
+)
+from transformers.cache_utils import Cache
+from QEfficient.transformers.cache_utils import QEffDynamicCache
+from QEfficient.transformers.modeling_outputs import QEffMoeModelOutputWithPast, QEffMoeCausalLMOutputWithPast
+from QEfficient.transformers.modeling_attn_mask_utils import _qeff_prepare_4d_causal_attention_mask
+
+
+def load_balancing_loss_func(gate_logits: torch.Tensor, num_experts: torch.Tensor = None, top_k=2) -> float:
```

Comment:
Is there any change in this function required? Else we can just call it and skip adding it here.